### DISCUSSION TITLE
Celery update to address CVE-2021-23727

### DISCUSSION THREAD

**Discussion by ghubz** [WeblateOrg/weblate#7076] [General] [answered]
Hi,

Is it safe to upgrade celery to the latest version? Versions < 5.2 are vulnerable to Stored Command Injection  as per https://github.com/celery/celery/blob/master/Changelog.rst

Thanks

**Comment by nijel** [comment, marked as answer]
It should be safe to upgrade.

Please note that the vulnerability is not exploitable without access to the Celery results backend (for Weblate it is most likely redis). That would indicate severe security issue in the infrastructure, and the attacker would be most likely able to exploit that in simpler means than via CVE-2021-23727.

**Reply by ghubz** [reply]
Thanks for the prompt reply and explanation!

**Reply by nijel** [reply]
Also, Weblate 4.10 officially supports Celery 5.2.x:

https://github.com/WeblateOrg/weblate/blob/e93c81ddb6354a7590b79694d11feab0cc16a2f7/requirements.txt#L3
